test/compressed_file_reader-spec.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const FileDB = require('../asset/compressed_file_reader/filedb');
+const fixtures = require('jest-fixtures');
+const fse = require('fs-extra');
+const path = require('path');
+
+describe('The compressed_file_reader', () => {
+    describe('when single file uploaded', () => {
+        const file = 'package.json.lz4';
+        let uploadDir;
+        let upload;
+        let workDir;
+        let decompress;
+        let ready;
+        let archive;
+        let filedb;
+        it('should be able to initialize', async () => {
+            uploadDir = await fixtures.copyFixtureIntoTempDir(__dirname, 't1');
+            upload = path.join(uploadDir, file);
+            workDir = await fixtures.createTempDir();
+            decompress = path.join(workDir, 'decompress', file);
+            ready = path.join(workDir, 'slice', file);
+            archive = path.join(workDir, 'archive', file);
+            filedb = await FileDB(workDir, path.resolve('asset'));
```

Comment:
I did this because `describe('...', async () => {})` isn't possible.  This was best idea I had.  Too many dependencies from one `it()` block to another.